File: src/main/java/com/mohe/spring/entity/StringArrayConverter.java
package com.mohe.spring.entity;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

@Converter
public class StringArrayConverter implements AttributeConverter<List<String>, String[]> {
    
    @Override
    public String[] convertToDatabaseColumn(List<String> attribute) {
        if (attribute == null) {
            return null;
        }
        return attribute.toArray(new String[0]);
    }
    
    @Override
    public List<String> convertToEntityAttribute(String[] dbData) {
        if (dbData == null) {
            return Collections.emptyList();
        }
        return Arrays.asList(dbData);
    }
}

File: src/main/java/com/mohe/spring/entity/StringArrayConverter.kt
package com.mohe.spring.entity

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter

@Converter
class StringArrayConverter : AttributeConverter<List<String>, Array<String>> {
    
    override fun convertToDatabaseColumn(attribute: List<String>?): Array<String>? {
        return attribute?.toTypedArray()
    }
    
    override fun convertToEntityAttribute(dbData: Array<String>?): List<String> {
        return dbData?.toList() ?: emptyList()
    }
}